spec/mspec/lib/mspec/helpers/ruby_exe.rb

```diff
@@ -132,7 +132,7 @@ def ruby_exe(code = :not_given, opts = {})
 
   escape = opts.delete(:escape)
   if code and !File.exist?(code) and escape != false
-    tmpfile = tmp("rubyexe.rb")
+    tmpfile = tmp(+"rubyexe.rb")
```

Comment:
This feels like the `tmp` helper should use `+` or `dup`, so not every caller needs to change.

Reply:
Problem is it mutates the argument, and some callers expect that. So raising on frozen makes it easier to find the broken ones.

Reply:
> Problem is it mutates the argument, and some callers expect that

We should fix those callers then.
I can try to take a look this week.
It feels definitely wrong that a helper method mutates its argument.

The practical problem is it's impossible to review the diff with all the `tmp(+` changes.

Reply:
As discussed on slack, in the end there was mostly one impacted callsite. So I reverted all that and just changed that one callsite a bit.